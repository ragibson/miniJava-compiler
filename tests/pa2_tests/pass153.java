// PA1 parse new pass
class Foo {
   void bar() {
      int[] newarr = new int[20];
   }
}


