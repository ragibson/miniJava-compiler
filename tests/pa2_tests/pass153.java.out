********** miniJava Compiler (PA2) **********
Syntactic Analysis ... 
Syntactic Analysis OK
======= AST Display =========================
Package
  ClassDeclList [1]
  . ClassDecl
  .   "Foo" classname
  .   FieldDeclList [0]
  .   MethodDeclList [1]
  .   . (public) MethodDecl
  .   .   VOID BaseType
  .   .   "bar" methodname
  .   .   ParameterDeclList [0]
  .   .   StmtList [1]
  .   .   . VarDeclStmt
  .   .   .   VarDecl
  .   .   .     ArrayType
  .   .   .       INT BaseType
  .   .   .     "newarr" varname
  .   .   .   NewArrayExpr
  .   .   .     INT BaseType
  .   .   .     LiteralExpr
  .   .   .       "20" IntLiteral
=============================================
